graphql_compiler/query_formatting/common.py

```diff
@@ -152,16 +133,25 @@ def deserialize_json_argument(
             GraphQLBoolean: bool
             GraphQLID: str
     """
-    try:
-        return _deserialize_anonymous_json_argument(strip_non_null_from_type(expected_type), value)
-    except (ValueError, TypeError) as e:
-        raise GraphQLInvalidArgumentError("Error parsing argument {}: {}".format(name, e))
+    stripped_type = strip_non_null_from_type(expected_type)
+
+    if isinstance(stripped_type, GraphQLList):
+        if not isinstance(value, list):
+            _raise_invalid_type_error(name, (list,), value)
+
+        inner_stripped_type = strip_non_null_from_type(stripped_type.of_type)
+        return [
+            _deserialize_json_scalar_argument(name, inner_stripped_type, element)
+            for element in value
+        ]
+    else:
+        return _deserialize_json_scalar_argument(name, stripped_type, value)
 
 
 def deserialize_multiple_json_arguments(
     arguments: Mapping[str, Any],
     expected_types: Mapping[str, Union[GraphQLNonNull[GraphQLScalarType], GraphQLScalarType]],
-) -> Mapping[str, Any]:
+) -> Dict[str, Any]:
```

Comment:
No reason to not be more specific about the return type.

Reply:
I realize this is not part of the PR, but `Deserialize GraphQL arguments parsed from a json file.` seems odd, specifically the `json file` bit. Why does it have to be a file? This function doesn't do anything with files, as far as I can tell.